kmir/src/kmir/kdist/mir-semantics/ty.k

```diff
@@ -46,283 +51,332 @@ syntax MIRString ::= mirString(String) [group(mir-string), symbol(MIRString::Str
+                         | boundRegionKindBrNamed(BrNamedDef, MIRString) [group(mir-enum), symbol(BoundRegionKind::BrNamed)]
+                         | "boundRegionKindBrEnv"                        [group(mir-enum), symbol(BoundRegionKind::BrEnv)]
+syntax BoundVariableKind ::= boundVariableKindTy(BoundTyKind)            [group(mir-enum), symbol(BoundVariableKind::Ty)]
+                           | boundVariableKindRegion(BoundRegionKind)    [group(mir-enum), symbol(BoundVariableKind::Region)]
+                           | "boundVariableKindConst"                    [group(mir-enum), symbol(BoundVariableKind::Const)]
+syntax BoundVariableKindList ::= List {BoundVariableKind, ""} [group(mir-list)]
+
+syntax FnSig ::= fnSig(inputsAndOutput: Tys, cVaradic: MIRBool, unsafety: Safety, abi: Abi) [group(mir---inputsAndOutput--cVariadic--unsafety--abi)]
+syntax BinderForFnSig ::= binderForFnSig(value: FnSig, boundVars: BoundVariableKindList)    [group(mir---value--boundVars)]
+syntax PolyFnSig ::= BinderForFnSig                                                         [group(mir)]
 // Not needed this way. We could just do PolyFnSig ::= binderForFnSig(value: FnSig, boundVars: BoundVariableKindList).
 
-syntax Ty ::= ty(MIRInt, TyKind) [symbol(ty)]
-            | ty(Int)            [group(mir-int)]
-syntax Tys ::= List {Ty, ""}
-
-syntax Pattern ::= patternRange(start: MaybeTyConst, end: MaybeTyConst, includeEnd: MIRBool)
-
-syntax TyConst ::= tyConst(kind: TyConstKind, id: TyConstId)
-syntax TyConstId ::= tyConstId(Int)
-syntax MaybeTyConst ::= someTyConst(TyConst) | "noTyConst"
-
-syntax MirConst ::= mirConst(kind: ConstantKind, ty: Ty, id: MirConstId)
-syntax MirConstId ::= mirConstId(Int)
-
-syntax TyConstKind ::= tyConstKindParam(ParamConst)
-                     | tyConstKindBound(DebruijinIndex, BoundVar)
-                     | tyConstKindUnevaluated(ConstDef, GenericArgs)
-                     | tyConstKindValue(Ty, Allocation)
-                     | tyConstKindZSTValue(Ty)
-
-syntax DebruijinIndex ::= debruijinIndex(Int)
-syntax UniverseIndex ::= universeIndex(Int)
-syntax EarlyParamRegion ::= earlyParamRegion(index: MIRInt, name: Symbol)
-syntax BoundVar ::= boundVar(Int)
-syntax BoundRegion ::= boundRegion(var: BoundVar, kind: BoundRegionKind)
-syntax PlaceholderForBoundRegion ::= placeholderForBoundRegion(universe: UniverseIndex, bound: BoundRegion)
-syntax RegionKind ::= regionKindReEarlyParam(EarlyParamRegion)
-                    | regionKindReBound(DebruijinIndex, BoundRegion)
-                    | "regionKindReStatic"
-                    | regionKindRePlaceholder(PlaceholderForBoundRegion)
-                    | "regionKindReErased"
-syntax Region ::= region(kind: RegionKind)
+// the struct variant of Ty was not observed in the wild yet, commenting it out
+syntax Ty ::= // ty(MIRInt, TyKind) [group(mir-enum), symbol(Ty::ty)]
+              ty(Int)            [group(mir-int)]
+
+syntax Tys ::= List {Ty, ""}     [group(mir-list)]
```

Comment:
List symbols missing. I noted one, but there are others. Generally, this will not make parsing fail, unless the specific rule is needed, so it can be an ongoing process.